### PR TITLE
S3 and Github unittests. 

### PR DESCRIPTION
**Pull request by chalenge**
**Related issues**
Relates to #14. See also #29 and #28
Adding unittest for s3 apps and github apps

**Describe the proposed solution**
To be merged after #28 to add more trust to our app creation process.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### CODE REVIEW

File: apps/ec2-old-instances/app_test.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+from unittest.mock import patch
+
+
+class EC2OldInstances(unittest.TestCase):
+    def setUp(self):
+        self.event = {
+            "farosToken": "farosToken",
+            "params": {"num_days": 2}
+        }
+        self.now = datetime.utcnow()
```
[the remark this answers is not in the document]
Comment (chalenge) [reply apps/ec2-old-instances/app_test.py:16]: This is mostly coz the apps are comparing the the values to `datetime.utcnow()` hence creating test dates that are relative to that. Constant dates won't work for cases that are in the future but within a time range.

File: apps/s3-missing-bucket-policy/app.py [hunk 2]
```diff
@@ -2,21 +2,25 @@
...
-        missing = list(frozenset(required_policy_statements).difference(attached))
+        missing = list(frozenset(required_policies).difference(attached))
         return missing
 
 
 def lambda_handler(event, context):
     client = FarosClient.from_event(event)
+    required_policy_param = event["params"].get("required_policy_statement")
+    if not required_policy_param:
+        raise KeyError(("Required policy statements not supplied. You can "
```
Comment (mtth) [inline apps/s3-missing-bucket-policy/app.py:19]: Omit the inner parens pair?

File: apps/s3-missing-bucket-policy/app.py [hunk 3]
```diff
@@ -2,21 +2,25 @@
...
+                        "define them with -p required_policy_statement="
+                        "<comma-separated-policies>"))
+    else:
```
Comment (mtth) [inline apps/s3-missing-bucket-policy/app.py:22]: Let's ditch this `else` to reduce nesting.

File: apps/ec2-old-instances/app_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtth) [reply apps/ec2-old-instances/app_test.py]: Understood, OK for now then. Would be good to add a (low priority) action item to improve the testability of these functions so we can use constants more easily.

**Pull request merged**